src/lib/io/include/sol-uart.h

```diff
@@ -224,21 +239,35 @@ struct sol_uart *sol_uart_open(const char *port_name, const struct sol_uart_conf
 void sol_uart_close(struct sol_uart *uart);
 
 /**
- * @brief Perform a UART asynchronous transmission.
+ * @brief Perform an UART asynchronous transmission.
+ *
+ * @param uart The UART bus handle
+ * @param blob The blob to be written
+ * @return 0 on success negative errno on error
+ *
+ */
+int sol_uart_write(struct sol_uart *uart, struct sol_blob *blob);
+
+/**
+ * @brief Perform an UART synchronous read
+ *
+ * @param uart The UART bus handle
+ * @param rx Where to store the UART data
+ * @param length number of bytes to read
+ *
+ * @return the number of bytes read or negative errno on error
+ */
+int sol_uart_read(struct sol_uart *uart, struct sol_buffer *buf);
+
+
+/**
+ * @brief Get the number of bytes that were not written yet.
  *
  * @param uart The UART bus handle
- * @param tx The output buffer to be sent
- * @param length number of bytes to be transfer
- * @param tx_cb callback to be called when transmission finish, in case of
- * success the status parameter on tx_cb should be equal to length otherwise
- * an error happen during the transmission
- * @param data the first parameter of tx_cb
- * @return true if transfer was started
- *
- * @note Caller should guarantee that tx buffer will not be freed until
- * callback is called.
- */
-bool sol_uart_write(struct sol_uart *uart, const uint8_t *tx, unsigned int length, void (*tx_cb)(void *data, struct sol_uart *uart, uint8_t *tx, int status), const void *data);
+ * @param pending_bytes Where to store the pending bytes amount
```

Comment:
should we document this is the sum of all pending blobs? or that's internal detail?